lib/core/src/packStruct.cpp

```diff
@@ -1917,19 +1919,18 @@ unpackXmlString( const void *&inPtr, packedOutput_t &unpackedOutput, int maxStrL
             return USER_PACKSTRUCT_INPUT_ERR;
         }
         else {
-            extendPackedOutput( unpackedOutput, myStrlen, outPtr );
+            extLen = myStrlen;
         }
     }
-    else {
-        extendPackedOutput( unpackedOutput, maxStrLen, outPtr );
-    }
 
-    if ( myStrlen > 0 ) {
-        strncpy( static_cast<char*>(outPtr), strBuf, myStrlen );
-        outStr = static_cast<char*>(outPtr);
-        outPtr = static_cast<char*>(outPtr) + myStrlen;
+    if ( SYS_MALLOC_ERR != extendPackedOutput( unpackedOutput, extLen, outPtr ) ) {
```

Comment:
Would it be better to return SYS_MALLOC_ERR on failure instead of continuing on? There are several "Argument with 'nonnull' attribute passed null" bugs in this file that are almost all related to this situation

Reply:
I would agree that you should return SYS_MALLOC_ERR if that is the result of extendedPackedOutput().  Can that function return any other possible error?  Why not capture the error in a variable and test and return that?

Reply:
extendedPackedOutput() can only return 0 or SYS_MALLOC_ERR. I will add a variable to capture the error. Thanks!